app/models/amusement_park.rb

```diff
@@ -1,3 +1,8 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+  has_many :mechanics, through: :rides
+
+  def all_mechanics
+    mechanics.distinct
```

Comment:
Nicely done! With this additional association on line 3, it's wayyyy easier to grab the mechanics. Without the association, you could do a join with a where clause that only grabs mechanics associated to rides from this park